### PR TITLE
Add cupyx.scipy.interpolate.CubicSpline

### PR DESCRIPTION
**Pull request by ev-br**
cross-ref https://github.com/cupy/cupy/issues/7186

The functionality is almost but not completely covered by `make_interp_spline`, so delegate to it whenever works. The edge cases of n=2, 3 are basically copy-paste of what the scipy counterpart does.
  
The LOC count in this PR is mostly gymnastics needed to retrofit a factory function into an LSP-breaking subclass.

### CODE REVIEW

File: cupyx/scipy/interpolate/_cubic.py [hunk 1]
```diff
@@ -412,3 +415,194 @@ def from_spline(cls, tck, extrapolate=None):
...
+    # redo the axis reshuffle in _PPolyBase.__init__:
+    # https://github.com/scipy/scipy/blob/v1.12.0/scipy/interpolate/_interpolate.py#L826
+    cvals_ = cupy.moveaxis(cvals, 0, axis+1)
+    cvals_ = cupy.moveaxis(cvals_, 0, axis+1)
+
+    pp = PPoly(cvals_, t, axis=axis)
+    return pp
+
+
+class CubicSpline(CubicHermiteSpline):
```
Comment (takagi) [inline cupyx/scipy/interpolate/_cubic.py:512]: Would you add its docstring?

Comment (ev-br) [reply cupyx/scipy/interpolate/_cubic.py:512]: Indeed. Done.

Comment (takagi) [reply cupyx/scipy/interpolate/_cubic.py:512]: Thanks!

**Comment by takagi** [comment]
Thanks, @ev-br!

**Pull request merged**